Form/Type/ModelType.php

```diff
@@ -214,9 +214,13 @@ public function configureOptions(OptionsResolver $resolver)
                     $valueProperty = $options['property'];
                     /** @var ModelCriteria $query */
                     $query = $options['query'];
-                    $getter = 'get' . ucfirst($query->getTableMap()->getColumn($valueProperty)->getPhpName());
 
-                    $choiceLabel = function($choice) use ($getter) {
+                    $choiceLabel = function($choice) use ($valueProperty) {
+                        $getter = 'get'.ucfirst($valueProperty);
+                        if (!method_exists($choice, $getter)) {
+                            $getter = 'get' . ucfirst($query->getTableMap()->getColumn($valueProperty)->getPhpName());
```

Comment:
@peter17, are you sure this is working? $query is not available within the closure, isn't it?